sdk/tests/DeserializationTests.cc

```diff
@@ -0,0 +1,254 @@
+  [[nodiscard]] inline const AccountId& getTestAccountIdTo() const { return mAccountIdTo; }
+  [[nodiscard]] inline const std::shared_ptr<ECDSAPublicKey>& getTestECDSAPublicKey() const { return mPublicKeyFromString; }
+
+private:
+  const uint64_t mShardNum = 1;
+  const uint64_t mRealmNum = 2;
+  const uint64_t mContractNum = 3;
+  const int32_t cents = 2;
+  const int32_t hbar = 1;
+  const uint64_t seconds = 100ULL;
+  const Hbar mBalance = Hbar(100LL);
+  const AccountId mAccountId = AccountId(0ULL, 0ULL, 10ULL);
+  const AccountId mAccountIdFrom = AccountId(4ULL);
+  const AccountId mAccountIdTo = AccountId(3ULL);
+  const std::unique_ptr<ECDSAPrivateKey> mPrivateKey = ECDSAPrivateKey::generatePrivateKey();
+  const std::shared_ptr<PublicKey> mPublicKeyFromPrivate = mPrivateKey->getPublicKey();
+  const std::shared_ptr<ECDSAPublicKey> mPublicKeyFromString = ECDSAPublicKey::fromString(mPublicKeyFromPrivate->toString());
+};
+
+TEST_F(DeserializationTests, DeserializeExchangeRateFromProtobufTest)
+{
+  // Given
+  const int32_t testCents = getTestCents();
+  const int32_t testHbar = getTestHbar();
+  const uint64_t testSeconds = getTestSeconds();
+
+  auto testProtoExchangeRate = std::make_unique<proto::ExchangeRate>();
+  proto::TimestampSeconds *testProtoExchangeRateSecs = testProtoExchangeRate->mutable_expirationtime();
+
+  testProtoExchangeRate->set_centequiv(testCents);
+  testProtoExchangeRate->set_hbarequiv(testHbar);
+  testProtoExchangeRateSecs->set_seconds(testSeconds);
+
+  // When
+  ExchangeRate exchangeRate = ExchangeRate::fromProtobuf(*testProtoExchangeRate);
+  
+  // Then
+  EXPECT_EQ(exchangeRate.getCurrentExchangeRate(), testCents / testHbar);
+  EXPECT_TRUE(exchangeRate.getExpirationTime().has_value());
+  EXPECT_EQ(exchangeRate.getExpirationTime().value().time_since_epoch().count(),
+            internal::TimestampConverter::fromProtobuf(*testProtoExchangeRateSecs).time_since_epoch().count());
+}
+
+TEST_F(DeserializationTests, DeserializeExchangeRateSetFromProtobufTest)
+{
+  // Given
+  const int32_t testCents = getTestCents();
+  const int32_t testHbar = getTestHbar();
+  const uint64_t testSeconds = getTestSeconds();
+
+  auto testProtoExchangeRate = std::make_unique<proto::ExchangeRate>();
+  testProtoExchangeRate->set_centequiv(testCents * testCents);
+  testProtoExchangeRate->set_hbarequiv(testHbar * testHbar);
+
+  proto::ExchangeRateSet testProtoExchangeRateSet;
+  testProtoExchangeRateSet.set_allocated_nextrate(testProtoExchangeRate.release());
+
+  // When
+  ExchangeRateSet exchangeRateSet = ExchangeRateSet::fromProtobuf(testProtoExchangeRateSet);
+  
+  // Then
+  EXPECT_FALSE(exchangeRateSet.getCurrentExchangeRate().has_value());
+  EXPECT_TRUE(exchangeRateSet.getNextExchangeRate().has_value());
+  EXPECT_EQ(exchangeRateSet.getNextExchangeRate().value().getCurrentExchangeRate(), testCents * testCents / testHbar * testHbar);
+}
+
+TEST_F(DeserializationTests, DeserializeTransactionReceiptFromProtobufTest)
```

Comment:
To write this test I used test [ProtobufTransactionReceipt](https://github.com/hashgraph/hedera-sdk-cpp/blob/c54139e8bc8f507e1f08d052b15eec3d188a0bd0/sdk/tests/TransactionReceiptTest.cc#L46) from [TransactionReceiptTest.cc](https://github.com/hashgraph/hedera-sdk-cpp/blob/main/sdk/tests/TransactionReceiptTest.cc).

Reply:
Removed test `DeserializeTransactionReceiptFromProtobufTest` in commit https://github.com/hashgraph/hedera-sdk-cpp/pull/176/commits/8f2579891c1a4a154cdf7b967d8ad881cd0791cc.